### PR TITLE
feat: set up CI test by Github Actions

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**
CI-test:
1. Run unit tests
2. Run linter

Formatted the python code to pass linter test.

### CODE REVIEW

**Comment by xiaozhenliu-gg5** [comment]
@software-dov mentioned in #4 that `black` would be a better choice for linter checking, is there special reason that it has advantages over flake8? I noticed that in micro-python, we use flake8 as linter lib.But I tried running `black` against the whole repo, and it's easy/clean to use, it helps formatting the code in place. And `--check` option can return only the check status without modification.
Thank you!

**Review by chrisdunelm — APPROVED** [review]
LGTM

File: src/comparator/message_comparator.py [hunk 1]
```diff
@@ -68,10 +68,10 @@ def _compareNestedMessages(self, nested_msg_dict_original, nested_msg_dict_updat
         message_name_original = nested_msg_dict_original.keys()
         message_name_update = nested_msg_dict_update.keys()
 
-        for msgName in set(message_name_original) - set (message_name_update):
+        for msgName in set(message_name_original) - set(message_name_update):
```
Comment (software-dov) [inline src/comparator/message_comparator.py:72]: Why recreate the original and update sets each time? Just make `message_name_original` and `meessage_name_update` sets when you define them above.

Comment (xiaozhenliu-gg5) [reply src/comparator/message_comparator.py]: 👍 Done, thank you!

**Pull request merged**